ColorCoding.c

```diff
@@ -0,0 +1,45 @@
+#include <stdio.h>
+#include <assert.h>
+#include "ColorCoding.h"
+
+void ColorPairToString(const ColorPair* colorPair, char* buffer) {
+    sprintf(buffer, "%s %s",
+        MajorColorNames[colorPair->majorColor],
+        MinorColorNames[colorPair->minorColor]);
+}
+
+ColorPair GetColorFromPairNumber(int pairNumber) {
+    ColorPair colorPair;
+    int zeroBasedPairNumber = pairNumber - 1;
+    colorPair.majorColor = 
+        (MajorColor)(zeroBasedPairNumber / numberOfMinorColors);
+    colorPair.minorColor =
+        (MinorColor)(zeroBasedPairNumber % numberOfMinorColors);
+    return colorPair;
+}
+
+int GetPairNumberFromColor(const ColorPair* colorPair) {
+    return colorPair->majorColor * numberOfMinorColors +
+            colorPair->minorColor + 1;
+}
+
+void colorCodingReferenceManual(void)
+{
+    int majorColorCount,minorColorCount;
+    ColorPair colorPairReference;
+    char colorPairManual[MAX_COLORPAIR];
+    printf("\n Color Coding Reference Manual\n");
+    printf("Major Color | \t Minor Color | \t Pair Number \n");
+    for(majorColorCount=0;majorColorCount<numberOfMajorColors;majorColorCount++)
+    {
+        for(minorColorCount=0;minorColorCount<numberOfMinorColors;minorColorCount++)
+        {
+            colorPairReference.majorColor = (MajorColor)majorColorCount;
+            colorPairReference.minorColor = (MinorColor)minorColorCount;
+            int pairNumberReference = GetPairNumberFromColor(&colorPairReference);
+            ColorPairToString(&colorPairReference, colorPairManual);
```

Comment:
Good use of tabs to align the fields. However, `ColorPairToString` does not have tabs. resulting in misaligned output. Try a test for that function to make it aligned.